core/src/main/java/org/infinispan/persistence/sifs/NonBlockingSoftIndexFileStore.java

```diff
@@ -704,33 +704,29 @@ private ByteBuffer toBuffer(byte[] array) {
             CloseableIterator::close);
    }
 
-   private <R> Flowable<R> handleFilePublisher(Flowable<Integer> filePublisher, boolean fetchValue, boolean fetchMetadata,
-                                               EntryFunctor<R> functor) {
-      return filePublisher.flatMap(f -> {
-         // Unbox here once
-         int file = f;
-         return Flowable.using(() -> {
-                  log.tracef("Loading entries from file %d", file);
-                  return Optional.ofNullable(fileProvider.getFile(file));
-               },
-               optHandle -> {
-                  if (!optHandle.isPresent()) {
-                     log.tracef("File %d was deleted during iteration", file);
-                     return Flowable.empty();
-                  }
+   private <R> Flowable<R> handleFilePublisher(int file, boolean fetchValue, boolean fetchMetadata,
+         EntryFunctor<R> functor) {
+      return Flowable.using(() -> {
+               log.tracef("Loading entries from file %d", file);
+               return Optional.ofNullable(fileProvider.getFile(file));
```

Comment:
@wburns wouldn't it have been simpler to check if `fileProvider.getFile(file) == null` outside `Flowable.using()` and return `Flowable.empty()`, so there's just one check instead of 3?

Reply:
Sure, that is how it should have been done before this PR even. The checks themselves are pretty tiny though, but the code should be better.